dataproc/src/test/java/ListClustersTest.java

```diff
@@ -0,0 +1,115 @@
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class ListClustersTest {
+
+  private static final String BASE_CLUSTER_NAME = "test-cluster";
+  private static final String REGION = "us-central1";
+  private static final String projectId = System.getenv("GOOGLE_CLOUD_PROJECT");
+
+  private String clusterName;
+  private String myEndpoint;
+  private ByteArrayOutputStream bout;
+  private PrintStream standardOutOrig;
+
+  private static void requireEnv(String varName) {
+    assertNotNull(
+        System.getenv(varName),
+        String.format("Environment variable '%s' is required to perform these tests.", varName));
+  }
+
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnv("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnv("GOOGLE_CLOUD_PROJECT");
+  }
+
+  @Before
+  public void setUp() throws IOException, InterruptedException {
+    clusterName = String.format("%s-%s", BASE_CLUSTER_NAME, UUID.randomUUID().toString());
+    bout = new ByteArrayOutputStream();
+    standardOutOrig = System.out;
+    System.setOut(new PrintStream(bout));
+    myEndpoint = String.format("%s-dataproc.googleapis.com:443", REGION);
+
+    ClusterControllerSettings clusterControllerSettings =
+        ClusterControllerSettings.newBuilder().setEndpoint(myEndpoint).build();
+
+    Cluster cluster =
+        Cluster.newBuilder()
+            .setClusterName(clusterName)
+            .setConfig(ClusterConfig.newBuilder().build())
+            .build();
+
+    try (ClusterControllerClient clusterControllerClient =
+        ClusterControllerClient.create(clusterControllerSettings)) {
+      OperationFuture<Cluster, ClusterOperationMetadata> createClusterAsyncRequest =
+          clusterControllerClient.createClusterAsync(projectId, REGION, cluster);
+      createClusterAsyncRequest.get();
+    } catch (ExecutionException e) {
+      System.out.println("[deleteCluster] Error during test cluster creation: \n" + e.toString());
+    }
+  }
+
+  @Test
+  public void ListClustersTest() throws IOException, InterruptedException {
+    ListClusters.ListClusters(projectId, REGION);
+    String output = bout.toString();
+
+    assertThat(output, CoreMatchers.containsString(clusterName));
+    assertThat(output, CoreMatchers.containsString("RUNNING"));
+  }
+
+  @After
+  public void teardown() throws IOException, InterruptedException {
+    System.setOut(standardOutOrig);
+    ClusterControllerSettings clusterControllerSettings =
+        ClusterControllerSettings.newBuilder().setEndpoint(myEndpoint).build();
+
+    try (ClusterControllerClient clusterControllerClient =
+        ClusterControllerClient.create(clusterControllerSettings)) {
+      OperationFuture<Empty, ClusterOperationMetadata> deleteClusterAsyncRequest =
+          clusterControllerClient.deleteClusterAsync(projectId, REGION, clusterName);
+      deleteClusterAsyncRequest.get();
+    } catch (ExecutionException e) {
```

Comment:
Don't catch these exceptions - JUNIT will handle errors and mark the test as failed correctly (the sysout message might get eatten up and ignored).